### PR TITLE
Added support for multipart form data.

### PR DESCRIPTION
**Pull request by BJap**

### CODE REVIEW

File: src/main/java/com/comcast/drivethru/api/HTTPRequestManager.java [hunk 1]
```diff
@@ -418,6 +441,17 @@ public Builder data(byte[] data)
             mData = data;
             return this;
         }
+        
+        /**
+         * Sets request's multipart data.
+         * @param multipart Multipart data to send the server
+         * @return {@link Builder} object
+         */
+        public Builder multipart(HttpEntity multipart)
```
Comment (dimaj) [inline src/main/java/com/comcast/drivethru/api/HTTPRequestManager.java:455]: Please add a `@see` annotation that will point to `MultipartEntityBuilder` api to help users create this multi-part entity

**Comment by dimaj** [comment]
Where are the unit tests?

File: src/main/java/com/comcast/drivethru/api/HTTPRequestManager.java (hunk 1 above)
Comment (BJap) [inline src/main/java/com/comcast/drivethru/api/HTTPRequestManager.java:455]: That was a worry of mine. Awesome point. I didn't think everyone would know how to use that builder to make an HttpEntity. That's a good place to throw them a bone

**Comment by BJap** [comment]
I was holding off on that until we solidify implementation and add it to this PR :smile:

File: src/main/java/com/comcast/drivethru/api/HTTPRequestManager.java [hunk 2]
```diff
@@ -86,10 +88,11 @@
      * Private constructor used by the Builder subclass to create and initialize the HTTPResposneManager object.
      * @param builder Builder object that has all the needed parts to make a HTTP(S) request
      */
-    private HTTPRequestManager(Builder builder)
+    HTTPRequestManager(Builder builder)
```
Comment (trentontrees) [inline src/main/java/com/comcast/drivethru/api/HTTPRequestManager.java]: So I see now why you made this change, but it shouldn't be publicly available...make it protected

**Pull request merged**